static/js/ui_init.js

```diff
@@ -213,7 +213,13 @@ exports.initialize_kitchen_sink_stuff = function () {
         }
         const row = event.msg_list.get_row(event.id);
         $(".selected_message").removeClass("selected_message");
-        row.addClass("selected_message");
+        const messagebox = $(row).children(".messagebox");
+        messagebox.addClass("selected_message");
+
+        if (!event.force_rerender) {
+            // Don't focus if local echos complete.
+            messagebox.focus();
+        }
```

Comment:
I'm not sure this is the only code path through which we rerender messages; we may want a check that's more like "Was this message previously the selected message"

Reply:
done